From ab40614c7ca0aae7fdbb96ceceb2464fafc5e1e4 Mon Sep 17 00:00:00 2001
From: YunYi Yang <yangyunyi2@huawei.com>
Date: Mon, 28 Aug 2023 16:23:40 +0800
Subject: [PATCH 125/256] fix kabi broken due to change of struct dev_link

driver inclusion
category: bugfix
bugzilla: https://gitee.com/src-openeuler/kernel/issues/I8F803

----------------------------------------------------------------------------

Commit ("drivers: base: Fix device link removal")
changes the struct dev_link, so we need to fix kabi broken problem.

Signed-off-by: YunYi Yang <yangyunyi2@huawei.com>
---
 include/linux/device.h | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/include/linux/device.h b/include/linux/device.h
index 21c11dad9cee..9c850a68fd2f 100644
--- a/include/linux/device.h
+++ b/include/linux/device.h
@@ -1098,10 +1098,13 @@ struct device_link {
 	u32 flags;
 	refcount_t rpm_active;
 	struct kref kref;
-	struct work_struct rm_work;
+#ifdef CONFIG_SRCU
+	struct rcu_head rcu_head;
+#endif
 	bool supplier_preactivated; /* Owned by consumer probe. */
 #ifndef __GENKSYMS__
 	struct device link_dev;
+	struct work_struct rm_work;
 #endif
 };
 
-- 
2.27.0

